pyproject.toml

```diff
@@ -37,62 +37,68 @@ dynamic = [
   "version",
 ]
 dependencies = [
-  "defusedcsv~=2.0.0",
-  "defusedxml~=0.7.1",
-  "Django~=3.2.20",
-  "django-allauth~=0.53.1",
-  "django-cleanup~=6.0.0",
+  "defusedcsv~=2.0",
+  "defusedxml~=0.7",
+  "django~=4.2",
+  "django-cleanup~=8.0",
   "django-compressor~=4.4",
-  "django-extensions~=3.2.0",
-  "django-filter~=21.1",
-  "django-libsass==0.9",
-  "django-mathfilters~=1.0.0",
-  "django-mptt~=0.13.4",
-  "django-rest-swagger~=2.2.0",
-  "django-settings-export~=1.2.1",
-  "django-widget-tweaks~=1.4.8",
-  "djangorestframework~=3.12.4",
-  "drf-extensions~=0.7.0",
-  "iso8601~=2.0.0",
-  "jsonfield~=3.1.0",
-  "Markdown~=3.3.7",
-  "pypandoc~=1.10.0",
-  "rules~=3.3.0",
+  "django-extensions~=3.2",
+  "django-filter~=23.2",
+  "django-libsass~=0.9",
+  "django-mathfilters~=1.0",
+  "django-mptt~=0.14",
+  "django-rest-swagger~=2.2",
+  "django-settings-export~=1.2",
+  "django-widget-tweaks~=1.5",
+  "djangorestframework~=3.14",
+  "drf-extensions~=0.7",
+  "iso8601~=2.0",
+  "jsonfield~=3.1",
+  "markdown~=3.4",
+  "pypandoc~=1.11",
+  "rules~=3.3",
 ]
 
 [project.optional-dependencies]
+allauth = [
+  "django-allauth~=0.56",
+]
 ci = [
-  "coveralls",
+  "coveralls~=3.3",
+  "rdmo[allauth]",
   "rdmo[dev]",
 ]
 dev = [
-  "django-debug-toolbar",
-  "pre-commit",
+  "django-debug-toolbar~=4.2",
+  "pre-commit~=3.4",
   "rdmo[pytest]",
 ]
 gunicorn = [
-  "gunicorn>=19.9",
+  "gunicorn~=21.2",
 ]
 ldap = [
-  "django-auth-ldap>=2",
+  "django-auth-ldap~=4.5",
 ]
 memcached = [
-  "python-memcached>=1.58",
+  "python-memcached~=1.5",
 ]
 mysql = [
-  "mysqlclient~=2.2.0",
+  "mysqlclient~=2.2",
 ]
 postgres = [
-  "psycopg2-binary~=2.9.6",
+  # once python 3.12 is officially released, the psycopg team will release a psycopg-binary for it.
+  # replace these two lines with "psycopg[binary]~=3.1"
+  "psycopg[binary]~=3.1; python_version<'3.12'",
```

Comment:
Python 3.8 - 3.11 with Django 4.2 can utilize psycopg3, i.e. psycopg[binary].

There will be a binary available, when Python 3.12 is officially released. I will adapt accordingly in October.